crates/astria-grpc-mock/src/mounted_mock.rs

```diff
@@ -0,0 +1,209 @@
+use std::sync::{
+    atomic::AtomicBool,
+    Arc,
+};
+
+use tokio::sync::Notify;
+use tonic::Request;
+
+use super::{
+    mock::{
+        Match as _,
+        Mock,
+    },
+    response::MockResponse,
+    AnyMessage,
+};
+use crate::{
+    clone_request,
+    clone_response,
+    response::ResponseResult,
+    verification::VerificationReport,
+};
+
+pub(crate) enum MockResult<U> {
+    NoMatch,
+    Success(tonic::Result<tonic::Response<U>>),
+    BadResponse(tonic::Status),
+}
+
+/// A wrapper of erased request to mock response.
+///
+/// Only used to implement `Clone` because [`Request`] doesn't.
+pub(crate) struct BadResponse {
+    pub(crate) request: Request<AnyMessage>,
+    pub(crate) mock_response: MockResponse,
+}
+
+impl BadResponse {
+    pub(crate) fn print(&self, mut buffer: impl std::fmt::Write, indent: &str) -> std::fmt::Result {
+        // TODO: Print the metadata map
```

Comment:
remove or implement?

Reply:
Leave for now. Gonna implement later because more work. I am going to make an issue.

Reply:
Added https://github.com/astriaorg/astria/issues/900 to the TODO comment